### PR TITLE
Make the class parameter mandatory for references

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Review by greg0ire — APPROVED** [review]

File: tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php [hunk 1]
```diff
@@ -96,7 +98,7 @@ public function testReferenceReconstruction(): void
 
         $roleFixture->load($em);
         // first test against managed state
-        $ref = $referenceRepository->getReference('admin-role');
+        $ref = $referenceRepository->getReference('admin-role', Role::class);
```
Comment (derrabus) [inline tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php:101]: Should we have made these adjustments to the tests on 1.x already?

**Comment by derrabus** [comment]
Please remember to update the UPGRADE.md file.

File: tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php (hunk 1 above)
Comment (VincentLanglet) [reply tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php:101]: Yes and no, depends on the test strategy.
The "old way" to call getReference must be tested on 1.x too.

On 1.x test should have been duplicated with
- The old way, working but triggering deprecation, and a `@group legacy`
- The new way working without deprecation.

Seems like no test were added/duplicated in https://github.com/doctrine/data-fixtures/pull/409, so only the old way were tested. Now on 2.x tests are updated.

**Review by SenseException — APPROVED** [review]

File: tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php (hunk 1 above)
Comment (derrabus) [reply tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php:101]: So, we're missing coverage for the non-deprecated way in 1.x currently? We should fix that first, I suppose.

Comment (VincentLanglet) [reply tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php:101]: I did https://github.com/doctrine/data-fixtures/pull/465

**Comment by VincentLanglet** [comment]
Friendly ping @derrabus @greg0ire ; can it be merged ?

**Review by derrabus — APPROVED** [review]

**Comment by ThomasLandauer** [comment]
This change is not documented at https://symfony.com/bundles/DoctrineFixturesBundle/current/index.html

**Comment by VincentLanglet** [comment]
> This change is not documented at [symfony.com/bundles/DoctrineFixturesBundle/current/index.html](https://symfony.com/bundles/DoctrineFixturesBundle/current/index.html)

DoctrineFixtureBundle only support data-fixture v2 since few hours.
PR need to be done to update https://github.com/doctrine/DoctrineFixturesBundle/blob/3.7.x/docs/index.rst

**Comment by derrabus** [comment]
Please open an issue (or a PR) on the bundle's repository.

**Pull request merged**